### ISSUE TITLE
Comments apply globally, affecting other files in strange ways

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#806]
This might be a newb question, but I'm currently in the process of migrating JSHint -> ESLint and one thing which caught me off-guard is that the comments to change a rule do not apply on a block level or even file level like they do in JSHint, but globally, ...or something.

Here's a reduced example to show the ~~bug~~ unexpected behaviour. No custom rules in play here, no .eslintrc files in the ancestry, just eslint 0.5.0 on Node 0.10.25.

Directory contains two files: `foo.js` and `bar.js`.

``` javascript
// foo.js
function foo() { return 1; }

// bar.js
function bar() { return 2; }
```

Running `eslint` in that directory gives this output:

```
bar.js
  1:0  error  Missing "use strict" statement  strict

foo.js
  1:0  error  Missing "use strict" statement  strict
```

...as expected.

But when I change `bar.js` to this code...

``` javascript
/*eslint strict: 0*/
function bar() { return 2; }
```

I get no warnings (exit code 0) from eslint.

What I expected was that it would still error on the missing 'use strict' in `foo.js`, since that doesn't include it.

Stranger still is if, instead of changing `bar.js`, `foo.js` is changed to include `/*eslint strict: 0 */`, then running eslint just gives a _warning_ for missing 'use-strict' in `bar.js`.

I assume this would have something to do with the order in which the files are checked.

---

Is there a way to temporarily (meaning, scoped by lines, scope or file) to modify the rules? As an example of how it might work, jslint and jshint allow code like this:

``` javascript
/*jslint bitwise: false */
x = y & 3; // no error
/*jslint bitwise: true */
z = w & 3; // error
```
